rotkehlchen/tests/unit/decoders/test_weth.py

```diff
@@ -465,3 +490,581 @@ def test_wxdai_wrap(database, gnosis_inquirer, gnosis_accounts):
+
+
+def test_weth_is_supported():
+    """Check that weth is supported for all the evm chains with ETH"""
+    assert (
+        set(CHAIN_ID_TO_WETH_MAPPING.keys()) ==
+        set(typing.get_args(SUPPORTED_CHAIN_IDS)) - CHAINS_WITHOUT_NATIVE_ETH
+    )
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('polygon_pos_accounts', [['0x33C0Aae5b2b6Eae2a6286B3a6621B55DcC02dC9e']])
+def test_wmatic_deposit_polygon_pos(database, polygon_pos_inquirer, polygon_pos_accounts):
+    evmhash = deserialize_evm_tx_hash(val='0xba581391d417a6dcc31031f1cf7cba6e63b701a8680828445ffdde73777843e1')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=polygon_pos_inquirer,
+        database=database,
+        tx_hash=evmhash,
+    )
+    timestamp = TimestampMS(1712851902000)
+    amount, gas_fees = '119.97566999849747', '0.007112105381183941'
+    expected_events = [
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.POLYGON_POS,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_POLYGON_POS_MATIC,
+            balance=Balance(amount=FVal(gas_fees)),
+            location_label=polygon_pos_accounts[0],
+            notes=f'Burned {gas_fees} MATIC for gas',
+            counterparty=CPT_GAS,
+        ),
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.POLYGON_POS,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=A_POLYGON_POS_MATIC,
+            balance=Balance(amount=FVal(amount)),
+            location_label=polygon_pos_accounts[0],
+            notes=f'Wrap {amount} MATIC in WMATIC',
+            counterparty=CPT_WMATIC,
+            address=string_to_evm_address('0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270'),
+        ),
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=2,
+            timestamp=timestamp,
+            location=Location.POLYGON_POS,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=A_WMATIC,
+            balance=Balance(amount=FVal(amount)),
+            location_label=polygon_pos_accounts[0],
+            notes=f'Receive {amount} WMATIC',
+            counterparty=CPT_WMATIC,
+            address=string_to_evm_address('0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270'),
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('polygon_pos_accounts', [['0xdAA9E3CA7500d7Ba3855dF9d8BCCde229C13919e']])
+def test_wmatic_withdraw_polygon_pos(database, polygon_pos_inquirer, polygon_pos_accounts):
+    evmhash = deserialize_evm_tx_hash(val='0xe90ed71875ff44ea45ea960d006ec4c0ccb86506cba494471aba4ba9dc86123f')  # noqa: E501
```

Comment:
For single arg functions no need to use kwargs. Especially since this makes it longer. Please change this in all tests you touched.

```suggestion
    evmhash = deserialize_evm_tx_hash('0xe90ed71875ff44ea45ea960d006ec4c0ccb86506cba494471aba4ba9dc86123f')  # noqa: E501
```